src/components/GroupsPanelForm/GroupsPanelForm.test.tsx

```diff
@@ -0,0 +1,211 @@
+import { screen, waitFor, within } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { setupServer } from "msw/node";
+import { vi } from "vitest";
+
+import {
+  getGetGroupsMockHandler,
+  getGetGroupsMockHandler400,
+  getGetGroupsResponseMock,
+  getGetGroupsResponseMock400,
+} from "api/groups/groups.msw";
+import { Label as GroupsPanelFormLabel } from "components/GroupsPanelForm";
+import { hasNotification, renderComponent } from "test/utils";
+
+import GroupsPanelForm from "./GroupsPanelForm";
+
+const mockApiServer = setupServer(
+  getGetGroupsMockHandler(
+    getGetGroupsResponseMock({
+      data: [
+        { id: "group1", name: "global" },
```

Comment:
It might be nice to add a mock function for groups like I've started doing for other entities e.g. https://github.com/canonical/rebac-admin/blob/main/src/mocks/roles.ts#L5